redfish-core/lib/pcie_slots.hpp

```diff
@@ -139,13 +139,68 @@ inline void
             "xyz.openbmc_project.Inventory.Item.PCIeDevice"});
 }
 
+inline void
+    linkAssociatedProcessor(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& pcieSlotPath, size_t index)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, pcieSlotPath,
+         index](const boost::system::error_code ec,
+                const std::variant<std::vector<std::string>>& endpoints) {
+            if (ec)
+            {
+                if (ec.value() == EBADR)
+                {
+                    // This PCIeSlot have no processor association.
+                    BMCWEB_LOG_DEBUG << "No processor association found";
```

Comment:
is this really necessary?  It seems like this can be deduced by the fact that there is no link.

Reply:
I was only referring to the debug trace.  I wasn't trying to suggest you remove any branches.

Reply:
FWIW I think I find what you originally had to be better from a readability point of view.  But the logic is the same...so up to you if you want to change it back or not.

Reply:
I am not able to recollect what I had earlier. :)